sentry/src/main/java/io/sentry/AsyncHttpTransportFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.sentry;
+
+import io.sentry.transport.AsyncHttpTransport;
+import io.sentry.transport.IConnectionConfigurator;
+import io.sentry.transport.ITransport;
+import io.sentry.transport.RateLimiter;
+import io.sentry.util.Objects;
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.jetbrains.annotations.NotNull;
+
+/** Creates {@link AsyncHttpTransport}. */
+final class AsyncHttpTransportFactory implements TransportFactory {
+
+  @Override
+  public ITransport create(final @NotNull SentryOptions options) {
```

Comment:
should we add `throws IllegalArgumentException` in the method signature?

Reply:
I think we should not, what's the reasonable way to handle this exception here anyways?

Reply:
if we handle it down its fine, I was just curious cus the `TransportFactory` now is the way to go for custom transports right

Reply:
We don't handle it, the exception is propagated to the very top and in the end Sentry init will fail.

This PR is a PR to another branch, so once whole package is there we can take a look at it again.